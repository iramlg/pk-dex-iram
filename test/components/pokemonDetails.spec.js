const React = require('react');
const PokemonDetails = require('../../src/components/pokemonDetails');
const renderer = require('react-test-renderer');
const { shallow } = require('enzyme');

const mockPokemon = {
  name: 'name',
  weight: 50,
  height: 30,
  sprites: {
    front_default: 'front_default'
  },
  types: [
    {
      name: 'name',
      type: {
        name: 'type_name'
      }
    }
  ]
}

describe('<PokemonDetails />', () => {
  describe('render', () => {
    it('test rendering', () => {
      const component = renderer.create(<PokemonDetails clearPokemon={ jest.fn() } pokemon={ mockPokemon }  />).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('test render instance', () => {
      const handleClearPokemon = jest.fn();
      const localInstance = shallow(<PokemonDetails clearPokemon={ handleClearPokemon } pokemon={ mockPokemon }  />);
      localInstance.find('.back-button').simulate('click');
      expect(handleClearPokemon).toHaveBeenCalled();
    });
  });

  describe('Props', () => {
    it('types', () => {
      expect(Object.keys(PokemonDetails.propTypes).length).toBe(2);

      ['pokemon', 'clearPokemon'].forEach(prop => {
        expect(PokemonDetails.propTypes[prop]).toBeTruthy();
      });

    });
  });
});
