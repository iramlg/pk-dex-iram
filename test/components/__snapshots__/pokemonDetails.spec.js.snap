// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonDetails /> render test rendering 1`] = `
<div
  className="pokemon-details"
>
  <span
    className="back-button"
    onClick={[Function]}
  >
    BACK
  </span>
  <img
    alt="name"
    className="sprite"
    src="front_default"
  />
  <div
    className="more-info"
  >
    <p>
      NAME
    </p>
    <p>
      HT 
      30
    </p>
    <p>
      WT 
      50
    </p>
    <div
      className="pokemon-details"
    >
      <span>
        TYPE_NAME
      </span>
    </div>
  </div>
</div>
`;
